python/Ganga/Core/GangaRepository/GangaRepositoryXML.py

```diff
@@ -129,9 +129,11 @@ def rmrf(name, count=0):
     if os.path.isdir(name):
 
         try:
-            remove_name = name + "_" + str(time.time()) + '__to_be_deleted_'
-            os.rename(name, remove_name)
-            #logger.debug("Move completed")
+            remove_name = name
+            if not remove_name.endswith('__to_be_deleted'):
+                remove_name = name + "_" + str(time.time()) + '__to_be_deleted_'
```

Comment:
This class calls itself and once the file has been renames it's passed as the file to be deleted again.
There are files on AFS which survive multiple calls to be removed including (but not limited to) lockfiles